test/sanitizer_suppressions/tsan

```diff
@@ -1,24 +1,36 @@
 # ThreadSanitizer suppressions
 # ============================
-
-# fChecked is theoretically racy, practically only in unit tests
-race:CheckBlock
-
-# WalletBatch (unidentified deadlock)
-deadlock:WalletBatch
-
-# Intentional deadlock in tests
+deadlock:BerkeleyBatch::VerifyDatabaseFile
+deadlock:BerkeleyDatabase::Flush
+deadlock:BerkeleyDatabase::Rewrite
+deadlock:CWallet::BackupWallet
+deadlock:CWallet::LoadWallet
+deadlock:MaybeCompactWalletDB
 deadlock:TestPotentialDeadLockDetected
+deadlock:WalletBatch::ErasePool
+deadlock:WalletBatch::EraseTx
+deadlock:WalletBatch::FindWalletTx
+deadlock:WalletBatch::LoadWallet
+deadlock:WalletBatch::ReadPool
+deadlock:WalletBatch::WalletBatch
+deadlock:WalletBatch::WriteBestBlock
+deadlock:WalletBatch::WriteCryptedKey
+deadlock:WalletBatch::WriteCScript
+deadlock:WalletBatch::WriteKey
+deadlock:WalletBatch::WriteName
+deadlock:WalletBatch::WritePool
+deadlock:WalletBatch::WritePurpose
+deadlock:WalletBatch::WriteTx
+deadlock:WalletBatch::WriteWatchOnly
+race:BerkeleyDatabase::Flush
+race:BerkeleyEnvironment::Flush
```

Comment:
This seems like a lot, what about something like this instead:

```diff
diff --git a/test/sanitizer_suppressions/tsan b/test/sanitizer_suppressions/tsan
index 209c46f..4edf68c 100644
--- a/test/sanitizer_suppressions/tsan
+++ b/test/sanitizer_suppressions/tsan
@@ -14,9 +14,10 @@ deadlock:TestPotentialDeadLockDetected
 race:src/qt/test/*
 deadlock:src/qt/test/*
 
-# WIP: Unidentified suppressions to run the functional tests
-#race:zmqpublishnotifier.cpp
-#
+# External libraries
+deadlock:libdb
+race:libzmq
+

Reply:
Fixed!